test/test_models.py

```diff
@@ -23,6 +25,37 @@ def get_models_from_module(module):
     return [v for k, v in module.__dict__.items() if callable(v) and k[0].lower() == k[0] and k[0] != "_"]
 
 
+@pytest.fixture
+def disable_weight_loading(mocker):
+    starting_point = models
+    python_module_names = {
+        info.name for info in pkgutil.walk_packages(starting_point.__path__, f"{starting_point.__name__}.")
+    }
+
+    function_name = "load_state_dict_from_url"
+    method_name = "load_state_dict"
+    targets = {f"torchvision._internally_replaced_utils.{function_name}", f"torch.nn.Module.{method_name}"}
+    for name in python_module_names:
+        python_module = sys.modules.get(name)
+        if not python_module:
+            continue
+
+        if function_name in python_module.__dict__:
+            targets.add(f"{python_module.__name__}.{function_name}")
+
+        targets.update(
+            {
+                f"{python_module.__name__}.{obj.__name__}.{method_name}"
+                for obj in python_module.__dict__.values()
+                if isinstance(obj, type) and issubclass(obj, nn.Module) and method_name in obj.__dict__
+            }
+        )
+
+    for target in targets:
+        with contextlib.suppress(AttributeError):
```

Comment:
There is (atleast) one situation where this patching doesn't work. We have the module `torchvision.models.alexnet` that defines the function `alexnet`, which is then imported into the `torchvision.models` namespace. Thus, after the import is complete, we can no longer access the module through `torchvision.models.alexnet`.

This is an anti-pattern and should be avoided. Given that everything we expose everything under `torchvision.models`, I suggest making all the other modules private. This avoids this naming issue and gives us the freedom to change the underlying structure however we want without needing to keep BC.

Reply:
Yeap, it is what it is. Not sure many other models have the same issue but I don't think it's worth breaking BC over it. Alexnet is in for historical reasons and unlikely to be used as a backbone for any modern architecture.